app/test-utils/testFunctions/questionTests/errorTests.js

```diff
@@ -0,0 +1,154 @@
+import { Selector, ClientFunction } from 'testcafe';
+import nock from 'nock';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const goToAnchorFromErrorSummary = ({
+  pageSetup,
+  sectionId,
+  questionId,
+  sectionApiUrl,
+  errorType,
+  sectionManifest,
+  apiLocalhost,
+}) => {
+  test(`should go to anchor when clicking the ${questionId} summary error link`, async (t) => {
+    await pageSetup({ t });
+    const questionType = sectionManifest.questions[questionId].type;
+    let modifiedQuestionId = questionId;
+    // bulletpoint-list numbers the input fields so it is not just the question id e.g. listing-1
+    if (questionType === 'bulletpoint-list') modifiedQuestionId = `${questionId}-1`;
+    nock(apiLocalhost)
+      .put(sectionApiUrl)
+      .reply(400, {
+        [errorType]: [modifiedQuestionId],
+      });
+
+    const errorSummary = Selector('[data-test-id="error-summary"]');
+    const errorSummaryList = Selector('.nhsuk-error-summary__list');
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+    await t
+      .expect(errorSummary.exists).notOk()
+      .click(submitButton.find('button'))
+      .expect(errorSummary.exists).ok()
+      .expect(errorSummaryList.find('li:nth-child(1) a').count).eql(1)
+      .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`#${modifiedQuestionId}`)
+      .click(errorSummaryList.find('li:nth-child(1) a'))
+      .expect(getLocation()).contains(`/solution/S100000-001/section/${sectionId}#${modifiedQuestionId}`);
+  });
+};
+
+const maxLengthErrorTest = ({
+  pageSetup,
+  sectionManifest,
+  questionId,
+  sectionApiUrl,
+  apiLocalhost,
+}) => {
+  test(`should show error summary and validation for ${questionId} question when it exceeds the maxLength`, async (t) => {
+    const questionType = sectionManifest.questions[questionId].type;
+    await pageSetup({ t });
+    nock(apiLocalhost)
+      .put(sectionApiUrl)
+      .reply(400, {
+        // bulletpoint-list numbers the input fields so it is not just the question id
+        maxLength: [questionType === 'bulletpoint-list' ? `${questionId}-1` : questionId],
```

Comment:
bit confused by this..

Reply:
this is because in a bulletpoint list the input field validation is in the format features-1, features-2 etc but it probably shouldn't be hard coded to 1 - I'll take a look

Reply:
ah yes ok that rings a bell. Grrr it's annoying that bulletpoint list is different to the other question types 😢

Reply:
I think given the number of different components we are running through this test generator there are bound to be one or 2 quirks like this!

Reply:
it is ok to be hard coded as that is what we are sending back from the nock respose

Reply:
I feel like I should say no. But go on then. I think this is a brilliant start and something we can look to touch up as we go along